pkg/api/api.go

```diff
@@ -136,7 +137,7 @@ func (hs *HTTPServer) registerRoutes() {
 
 			userRoute.Get("/auth-tokens", Wrap(hs.GetUserAuthTokens))
 			userRoute.Post("/revoke-auth-token", bind(models.RevokeAuthTokenCmd{}), Wrap(hs.RevokeUserAuthToken))
-		})
+		}, reqSignedInNonOAuth)
```

Comment:
This looks like it's applied too widely. There are a lot of endpoints in this group that shouldn't be deactivated when you want to prevent the user from setting a password.